react-form-controls/my-app/src/RegistrationFormUncontrolled.tsx

```diff
@@ -0,0 +1,27 @@
+export function RegistrationFormUncontrolled() {
+  function handleSubmit(e) {
```

Comment:
The proper way to type this is `e: FormEvent`, which you import from React.

Then you need to use `e.currentTarget` instead of `e.target` to get the types correct.